lib/middleware/websocket/controller.rb

```diff
@@ -6,21 +8,22 @@ def initialize
       @clients = []
     end
 
-    def on_open(client)
-      @clients << client
-      puts 'Websocket connection established on the server.'
-      client.write 'ack from the server'
+    def on_open(incoming_client)
+      @clients <<
+        Client.new(
+          connection_client: incoming_client, client_id: @clients.length + 1
```

Comment:
This will result in duplicate IDs, assuming we remove clients on disconnect.

1. `client1` connects, assigned ID 1
2. `client2` connects, assigned ID 2
3. `client2` disconnects
4. `client3` connects, assigned ID 2

Does `incoming_client` come in with any type of identifier?

Reply:
fixed for now, we will eventually implement some sort of unique identifier.